core/domain/state_domain.py

```diff
@@ -1787,9 +1786,7 @@ def update_interaction_hints(self, hints_list):
                 % hints_list)
         old_content_id_list = [
             hint.hint_content.content_id for hint in self.interaction.hints]
-        self.interaction.hints = [
-            Hint.from_dict(hint_dict)
-            for hint_dict in hints_list]
+        self.interaction.hints = hints_list[:]
```

Comment:
`copy.deepcopy(hints_list)` is the idiom we typically use, I think.

Reply:
Done